### PR TITLE
Upgrade 25th example to Elm 0.19

### PR DESCRIPTION
**Pull request by jhrcek**
Hello @bryanjenningz,
Thank you for this repo. It is a wonderful resource that I'd like to use in a workshop I'm organizing at work to introduce my colleagues to elm. I updated the 25th example to work with elm 0.19 and also replaced the elm-package.json with 0.19 compatible elm.json. I need that change because I'd like to be able to use this repo as-is from elm reactor (at least the first 20 examples) . I had to update most of the comments in the 25th example due to significant changes in how Browser.application works in elm 0.19. Feel free to edit those comments to fit better with your presentation style.

### CODE REVIEW

**Comment by jhrcek** [comment]
I hope you don't mind, but I also rewrote the README.md to markdown format, because that way it can be rendered nicely when you run `elm reactor` in the root of the project. The html markup that was used originally is rendered verbatim in the reactor and doesn't look good :-)

Here is how it looked in elm reactor before

screenshot from 2019-01-20 16-32-28

And after my changes .. 

screenshot from 2019-01-20 16-31-59

File: 25-navigation-todos.elm [hunk 1]
```diff
@@ -301,18 +351,29 @@ subscriptions model =
...
--- which returns the filter that we will use for that URL. So if the page's
--- URL is initially "/#completed", then (locationToFilter location) will return
--- Completed as the filter, so the filter value will be Completed, which will
--- make it show the completed todos are shown.
-
-
-init : Flags -> Navigation.Location -> ( Model, Cmd Msg )
-init flags location =
-    ( Model "" flags.todos Nothing (locationToFilter location)
+-- We're now using Browser.appication, whose init function requires two additional
```
Comment (bryanjenningz) [inline 25-navigation-todos.elm:354]: typo `application`

File: 25-navigation-todos.elm [hunk 2]
```diff
@@ -322,16 +383,32 @@ type alias Flags =
...
 
 
--- We're using Navigation.programWithFlags instead of Html.programWithFlags,
--- which takes UrlChange and it will pass the location to UrlChange whenever
--- the URL changes and then that will get passed into the update function.
+{-
+   We're using Browser.application instead of Browser.element, which extends the
+   latter in 3 important ways:
+
+   1. init gets two addition al parameters:
```
Comment (bryanjenningz) [inline 25-navigation-todos.elm:390]: typo `additional`

File: README.md [hunk 3]
```diff
@@ -1,62 +1,65 @@
...
-  <li><a href="https://github.com/rtfeldman/elm-spa-example">RealWorld Elm example app</a></li>
-</ul>
-
-<p>These 25 Elm examples were inspired by the <a href="http://elm-lang.org/examples">examples</a> on the Elm website and the <a href="https://guide.elm-lang.org/">official Elm guide</a>. Each example tries to build off of the previous example by adding a small amount of code so that it's easy to understand and see how you can build stuff in Elm. If you have any suggestions for more examples that should be added or if anything is unclear, add an issue or make a pull request.</p>
+```
+
+The above code will compile and open example 21 in your default browser.
+To compile and view examples 22 through 25 locally, you do the same process of
+compiling the Elm file you want to the elm.js JavaScript file,
+then opening the HTML file you want to view in your browser.
```
Comment (bryanjenningz) [inline README.md:49]: Line 49 can probably be replaced with this:
then running `elm reactor` and selecting the HTML file in your browser at localhost:8000.

**Comment by bryanjenningz** [comment]
Thank you so much for doing this! I really appreciate this. I tested your changes locally and it works! There are 2 small typos and 1 line in the README that I want to change, I'll merge this pull request in and then fix those typos for you. Thanks again!

**Pull request merged**